src/NodeVisitor/EscaperNodeVisitor.php

```diff
@@ -49,7 +50,7 @@ public function __construct()
     public function enterNode(Node $node, Environment $env): Node
     {
         if ($node instanceof ModuleNode) {
-            if ($env->hasExtension(EscaperExtension::class) && $defaultStrategy = $env->getExtension(EscaperExtension::class)->getDefaultStrategy($node->getTemplateName())) {
+            if ($env->hasExtension(EscaperExtension::class) && $defaultStrategy = $env->getRuntime(EscaperRuntime::class)->getDefaultStrategy($node->getTemplateName())) {
```

Comment:
should the default strategy be in the runtime ? It seems weird to me to use the runtime at compile time.

Reply:
Right. We don't even use it in the runtime, this is only needed at compilation time. See #4072